peer/test_utils.go

```diff
@@ -643,3 +529,200 @@ func (m *mockMessageConn) LocalAddr() net.Addr {
 func (m *mockMessageConn) Close() error {
 	return nil
 }
+
+type testReturnParams struct {
```

Comment:
Same here. This name gives us no intuition about what this is. Tell us instead what kind of setup this is. What are we testing with these? You could name it `PeerStorageTestHarness`